Super_Simple_RGB_WiFi_Lamp/Super_Simple_RGB_WiFi_Lamp.ino

```diff
@@ -52,6 +54,15 @@ String SSID = "";
 String Password = "";
 // ########################################################## End of Sketch Variables ##########################################################
 
+class ModeBase
+{
+public:
+    virtual void render();
+    virtual void applyConfig(JsonVariant& settings);
+};
+
+std::map<String, ModeBase*> modes;
```

Comment:
More just thinking out loud here: I like this a lot. I think map will be best container option currently. Only thoughts are if it were to become a dynamic list where you could add and remove modes would it be better as a list?